samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -71,12 +73,10 @@ public ExecutionPlan plan(OperatorSpecGraph specGraph) {
     JobGraph jobGraph = createJobGraph(specGraph);
 
     // fetch the external streams partition info
-    updateExistingPartitions(jobGraph, streamManager);
+    fetchInputAndOutputStreamPartitions(jobGraph, streamManager);
 
-    if (!jobGraph.getIntermediateStreamEdges().isEmpty()) {
-      // figure out the partitions for internal streams
-      calculatePartitions(jobGraph);
-    }
+    // figure out the partitions for internal streams
+    calculatePartitions(jobGraph);
```

Comment:
Did this method already handle the empty case ie., was the existing check just superficial?

Reply:
Yes, but the primary reason I removed this check is because we need join input validation even in the absence of any intermediate streams. This is why the ExecutionPlanner did not complain about the simple case of joining 2 input streams w/ different partition counts.